python/ray/dag/compiled_dag_node.py

```diff
@@ -127,14 +127,15 @@ def _exec_task(self, task: "ExecutableTask", idx: int) -> bool:
     res = None
     try:
         res = input_reader.begin_read()
-    except ValueError as exc:
-        # ValueError is raised if a type hint was set and the returned
-        # type did not match the hint.
+    except IOError:
+        # Channel closed. Exit the loop.
+        return True
+    except Exception as exc:
+        # Previous task raised an application-level exception.
+        # Propagate it and skip the actual task.
         output_writer.write(exc)
```

Comment:
Why does this not need _wrap_exception()?

Reply:
I believe this is because the exception has already been wrapped by the original task that errored. Will add a comment!

Reply:
Ah, I see. It's wrapped in a RayTaskError, which is itself an exception. So `exc` is an instance of RayTaskError here.